controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1422,6 +1432,9 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 	}
 	if provResult.ErrorMessage != "" {
 		info.host.Status.Provisioning.Firmware = nil
+		if hfcDirty {
+			hfc.Status.Updates = nil
```

Comment:
I don't think this change is going to be saved. On line 1434 we're relying on the higher-level controller code to save the BMH but nothing is gong to save the HFC after you return.

Changing the HFC here runs a risk that we override the error message with a transient failure from the unability to update the HFC. I wonder if we can do it earlier? Is there any harm in wiping the current Updates after line 1426 (provided that hfcDirty is true)?

Reply:
So maybe we have the same problem in https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller.go#L1183 ? 

`Changing the HFC here runs a risk that we override the error message with a transient failure from the unability to update the HFC. I wonder if we can do it earlier? Is there any harm in wiping the current Updates after line 1426 (provided that hfcDirty is true)?`

I would say tehre is a harm if we try to change the `Spec.Updates` from the `hfc` because we call the `r.saveHostFirmwareComponents(prov, info, hfc)` on L 1450

Reply:
Yes, that line is for sure doing nothing and so is this one. You'd have to call `saveHostFirmwareComponents()` after for it to have any effect.

> Changing the HFC here runs a risk that we override the error message with a transient failure

The error message will still be there when you come back.

Reply:
> Yes, that line is for sure doing nothing and so is this one. 

`that` line are you referring to what exactly?  `this one` I'm assuming you are talking about the `hfc.Status.Updates = nil`  L1440.

> You'd have to call `saveHostFirmwareComponents()` after for it to have any effect.

Ok, so we will need it twice the `saveHostFirmwareComponents()` because we have the return in the if block `	if provResult.ErrorMessage != "" {`

Reply:
Thanks, I've updated! to include the `r.Status().Update(info.ctx, hfc)`